src/custom/state/claim/hooks/index.ts

```diff
@@ -275,16 +276,18 @@ export function useUserClaims(account: Account, optionalChainId?: SupportedChain
   return { claims: claimKey ? claimInfo[claimKey] : null, isLoading }
 }
 
-let fetch_deployment_timestamp_promise: Promise<number> | null = null
-function fetchDeploymentTimestamp(vCowContract: VCowType) {
-  if (!fetch_deployment_timestamp_promise) {
-    fetch_deployment_timestamp_promise = vCowContract.deploymentTimestamp().then((ts) => {
+const FETCH_DEPLOYMENT_TIME_PROMISES: Map<ChainId, Promise<number>> = new Map()
+function fetchDeploymentTimestamp(vCowContract: VCowType, chainId: ChainId): Promise<number> {
+  let deploymentTimePromise = FETCH_DEPLOYMENT_TIME_PROMISES.get(chainId)
+
+  if (!deploymentTimePromise) {
+    deploymentTimePromise = vCowContract.deploymentTimestamp().then((ts) => {
       console.log(`Deployment timestamp in seconds: ${ts.toString()}`)
       return ts.mul('1000').toNumber()
     })
```

Comment:
I think after this line we can add `FETCH_DEPLOYMENT_TIME_PROMISES.set(chainId, deploymentTimePromise)` as part of that if block statement.

Reply:
just, good catch. Forgot to cache it!